### DISCUSSION TITLE
Read-only file system (30) :(

### DISCUSSION THREAD

**Discussion by 1o1o1** [transmission/transmission#8426] [Support]
### What is the issue?

Read-only file system (30)  but all rights for user transmission is open 

transmission-daemon[810184]: [2023-05-20 12:36:35.051] ERR open-files.cc:183 Couldn't open '/pub/incoming/.incomplete/Video.mkv.part': Read-only file system (30)
transmission-daemon[810184]: [2023-05-20 12:36:21.052]ERR open-files.cc:183 Couldn't open '/pub/incoming/.incomplete/Video.mkv.part': Read-only file system (30) (inout.cc:156)

Archlinux pkg  transmission-cli-4.0.3-1-aarch64.pkg.tar.xz

### Which application of Transmission?

transmission-daemon

### Which version of Transmission?

4.0.3

**Comment by 1o1o1** [comment]
may be because I have linked folder :   /pub -> /usr/public/pub

**Comment by 1o1o1** [comment]
downgraded to ver 3.00
working fine with the same config

**Comment by ckerr** [comment]
I don't understand the issue; please clarify -- is `/pub/incoming/.incomplete` read-only?

**Comment by 1o1o1** [comment]
can't start download  
"Read-only file system (30)"

drwxr-xr-x 2 transmission transmission        4096 may 20 13:31  .incomplete/
this is nonsense

Has anyone else come across this?

**Comment by ThePowerTool** [comment]
I have seen an issue with similar results.

1. Can you post the complete output of mount?
2. Can you post the parent directory listing (with attributes like you did, above)?
3. Can you post *one line (one file)* the output of "ls -l ~./.incomplete"

Do not post any personal information (e.g. if a file (or directory) appears as "file-your-name.txt" then change it to "random.txt" as names are not important; just the permissions (or in the case of the mount command, just the pointers).

**Comment by 1o1o1** [comment]
I don't know what the problem is!
I use ArchLinux with systemd service

Just adding BindPaths=/pub/incoming to transmission.service solved the issue!

[Unit]                                                                                                                                                
Description=Transmission BitTorrent Daemon                                                                                                            
Wants=network-online.target                                                                                                                           
After=network-online.target                                                                                                                           
                                                                                                                                                      
[Service]                                                                                                                                             
User=transmission                                                                                                                                     
Type=notify                                                                                                                                           
ExecStart=/usr/bin/transmission-daemon -f --log-level=error                                                                                           
ExecReload=/bin/kill -s HUP $MAINPID                                                                                                                  
NoNewPrivileges=true                                                                                                                                  
MemoryDenyWriteExecute=true                                                                                                                           
ProtectSystem=true                                                                                                                                    
PrivateTmp=true                                                                                                                                       
BindPaths=/pub/incoming                                                                                                                               
                                                                                                                                                      
[Install]                                                                                                                                             
WantedBy=multi-user.target

same as here
https://unix.stackexchange.com/questions/731364/transmission-cant-access-mount-read-only-file-system

**Comment by simplepad** [comment]
ProtectSystem=true mounts /usr/ as read only https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#ProtectSystem=
